controller/array_action/array_mediator_svc.py

```diff
@@ -185,14 +172,28 @@ def is_active(self):
 
     def _generate_volume_response(self, cli_volume):
         source_volume_wwn = self._get_source_volume_wwn_if_exists(cli_volume)
+        space_efficiency = self._get_cli_volume_space_efficiency(cli_volume)
         return Volume(
-            int(cli_volume.capacity),
-            cli_volume.vdisk_UID,
-            cli_volume.name,
-            self.endpoint,
-            cli_volume.mdisk_grp_name,
-            source_volume_wwn,
-            self.array_type)
+            vol_size_bytes=int(cli_volume.capacity),
+            vol_id=cli_volume.vdisk_UID,
+            vol_name=cli_volume.name,
+            array_address=self.endpoint,
+            pool_name=cli_volume.mdisk_grp_name,
+            copy_source_id=source_volume_wwn,
+            array_type=self.array_type,
+            space_efficiency=space_efficiency
+        )
+
+    def _get_cli_volume_space_efficiency(self, cli_volume):
```

Comment:
why was this move inside the class (`get_cli_volume_capabilities` was outside)?

Reply:
to get the volume, but now its not there, I will move it out again

Reply:
done